drivers/audio/wm8960.c

```diff
@@ -0,0 +1,752 @@
+/*
+ * Copyright (c) 2021 NXP Semiconductor INC.
+ * All rights reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <errno.h>
+
+#include <sys/util.h>
+
+#include <device.h>
+#include <drivers/i2c.h>
+#include <drivers/gpio.h>
+
+#include <audio/codec.h>
+
+#include <fsl_clock.h>
+
+#define LOG_LEVEL CONFIG_AUDIO_CODEC_LOG_LEVEL
+#include <logging/log.h>
+LOG_MODULE_REGISTER(wolfson_wm8960);
+
+#include "wm8960.h"
+
+#define DT_DRV_COMPAT wolfson_wm8960
+
+#define CODEC_OUTPUT_VOLUME_MAX 0
+#define CODEC_OUTPUT_VOLUME_MIN (-78 * 2)
+
+struct codec_driver_config {
+	const char *i2c_dev_name;
+	uint8_t i2c_address;
+	const char *gpio_dev_name;
+	uint32_t gpio_pin;
+	int gpio_flags;
+};
+
+struct codec_driver_data {
+	struct device *i2c_device;
+	struct device *gpio_device;
+	uint16_t reg_cache[WM8960_CACHEREGNUM];
+};
+
+static struct codec_driver_config codec_device_config = {
+	.i2c_dev_name = DT_INST_BUS_LABEL(0),
+	.i2c_address = DT_INST_REG_ADDR(0),
+};
+
+static struct codec_driver_data codec_device_data;
+
+static const uint16_t wm8960_reg[WM8960_CACHEREGNUM] = {
```

Comment:
It would be helpful if we could add a comment describing this array.

Reply:
ok